local-cloudbuild.py

```diff
@@ -0,0 +1,148 @@
+#!/usr/bin/env python
+#
+# Run the steps of cloudbuild.yaml locally, emulating Google Container
+# Builder functionality.  Does not actually push images to the
+# Container Registry.  Not all functionality is supported.
+#
+# Based on https://cloud.google.com/container-builder/docs/api/build-steps
+
+# System packages
+import argparse
+import getpass
+import os
+import shutil
+import subprocess
+import sys
+import tempfile
+
+# Third party packages
+import yaml
+
+
+# Types
+class CloudBuildError(Exception):
+    pass
+
+
+def main(argv):
```

Comment:
This file was an experiment in function ordering: preorder call-tree traversal, so the most important or most high-level functions come first, vs post-order where `main` would come last, or alphabetical order for when I'm really feeling ambitious, or vs the ever-popular random order that every source file eventually devolves to.

My conclusion: Not a big difference either way.  I'll move main to the end.